gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/RoleRepository.java

```diff
@@ -64,4 +64,7 @@ Set<Role> findByScopeAndReferenceIdAndReferenceType(RoleScope scope, String refe
         throws TechnicalException;
 
     Set<Role> findAllByReferenceIdAndReferenceType(String referenceId, RoleReferenceType referenceType) throws TechnicalException;
+
+    Role findByIdAndReferenceIdAndReferenceType(String roleId, String referenceId, RoleReferenceType referenceType)
```

Comment:
What about returning an `Optional<Role>`. It is usually the signature used for `find*` at least in the repository layer 🙈